src/codeflare_sdk/cluster/cluster.py

```diff
@@ -302,23 +393,42 @@ def _get_ray_clusters(namespace="default") -> List[RayCluster]:  # pragma: no co
 
 def _get_app_wrappers(
     namespace="default", filter=List[AppWrapperStatus]
-) -> List[AppWrapper]:  # pragma: no cover
+) -> List[AppWrapper]:
     list_of_app_wrappers = []
 
-    with oc.project(namespace), oc.timeout(10 * 60):
-        app_wrappers = oc.selector("appwrappers").objects()
+    try:
+        with oc.project(namespace), oc.timeout(10 * 60):
+            app_wrappers = oc.selector("appwrappers").objects()
+    except oc.OpenShiftPythonException as osp:  # pragma: no cover
+        error_msg = osp.result.err()
+        if (
+            'the server doesn\'t have a resource type "appwrappers"' in error_msg
+            or "forbidden" in error_msg
+            or "Unauthorized" in error_msg
+            or "Missing or incomplete configuration" in error_msg
+        ):
+            raise PermissionError(
+                "Action not permitted, have you put in correct/up-to-date auth credentials?"
+            )
+        else:
+            raise osp
 
     for item in app_wrappers:
         app_wrapper = _map_to_app_wrapper(item)
         if filter and app_wrapper.status in filter:
             list_of_app_wrappers.append(app_wrapper)
         else:
+            # Unsure what the purpose of the filter is
```

Comment:
I think that this would allow you to specify which types of statuses you want to get. 

But I also don't see where this function is used. It doesn't seem to get called elsewhere in this file.

Reply:
Yeah, my confusion is why it does the same exact thing in the if and the else, makes the if statement, and filter, useless (unless I'm just missing something)